### DISCUSSION TITLE
Scaling DB Instances and Concurrent Users-Connections Limit

### DISCUSSION THREAD

**Discussion by billypap1** [supabase/supabase#5975] [General]
Hello, I’m testing Supabase for my chat app but I’m not sure if it can handle many concurrent users. 
Has anyone tested it for this type of app?

So my questions are below:

1. what’s the limit on concurrent users (listeners) for the Pro plan (2GB RAM - 2vCPU instance)? Is it able to handle 1000 concurrent users?
2.  Is there a de facto number of concurrent users per resource’s ratio?
3. Are there benchmarks for real-time load?

PS. I use Supabase cloud and I want to use it only for Real-time and chat storage. The users are stored in Firebase.

**Comment by awalias** [comment]
Hey we have some performance data in this post: https://supabase.com/blog/2021/12/01/realtime-row-level-security-in-postgresql

**Reply by billypap1** [reply]
Hello, yes I have checked this post. In a previous conversation with support, they sent me a screenshot with the folowing instances in table in case  I want to upgrade later: 
1) 2vCPU - 2GB RAM (included in PAYG-PRO)
2) 2vCPU - 4GB RAM
3)  2vCPU - 8GB RAM
4)  4vCPU - 16GB RAM
5)  8vCPU - 32GB RAM

I would like to know, what's the limit on concurrent users for each instance size. Normal load, basic chat app. 
For example, can default PRO db Instance (2vCPU - 2GB RAM) handle 1K concurrent users?

**Reply by sumeone-somewhere** [reply]
Does it?

**Reply by jinsley8** [reply]
> Hello, yes I have checked this post. In a previous conversation with support, they sent me a screenshot with the folowing instances in table in case I want to upgrade later:
> 
> 1. 2vCPU - 2GB RAM (included in PAYG-PRO)
> 2. 2vCPU - 4GB RAM
> 3. 2vCPU - 8GB RAM
> 4. 4vCPU - 16GB RAM
> 5. 8vCPU - 32GB RAM
> 
> I would like to know, what's the limit on concurrent users for each instance size. Normal load, basic chat app. For example, can default PRO db Instance (2vCPU - 2GB RAM) handle 1K concurrent users?

Limits are here: https://supabase.com/docs/guides/platform/performance
